server/src/main/java/org/apache/iotdb/db/utils/TypeInferenceUtils.java

```diff
@@ -81,6 +81,8 @@ public static TSDataType getPredictedDataType(Object value, boolean inferType) {
           return floatingStringInferType;
         }
         // "NaN" is returned if the NaN Literal is given in Parser
+      } else if ("null".equals(strValue)) {
```

Comment:
What if user pass a "NULL" as value? Should we think "NULL" also a null?

Reply:
I'm going to let antlr transfer `NULL` to `null`. 
But this strValue may also come from session client by insertStringRecord, right? Yes, I think consider `NULL` here is necessary.